src/linear_algebra/affine_space/affine_map.lean

```diff
@@ -497,6 +497,18 @@ instance : module k (P1 →ᵃ[k] V2) :=
 
 @[simp] lemma coe_smul (c : k) (f : P1 →ᵃ[k] V2) : ⇑(c • f) = c • f := rfl
 
+@[simp] lemma smul_linear (t : k) (f : P1 →ᵃ[k] V2) : (t • f).linear = t • f.linear := rfl
+
+/-- The space of affine maps between two modules is linearly equivalent to the product of the
+domain with the space of linear maps. -/
+@[simps] def to_const_prod_linear_map : (V1 →ᵃ[k] V2) ≃ₗ[k] V2 × (V1 →ₗ[k] V2) :=
+{ to_fun    := λ f, ⟨f 0, f.linear⟩,
+  inv_fun   := λ p, const k V1 p.1 + p.2.to_affine_map,
```

Comment:
Given `decomp` is stated the other way, would it make sense to write
```suggestion
  inv_fun   := λ p, p.2.to_affine_map + const k V1 p.1,
```

Reply:
Yes, of course. Thanks!